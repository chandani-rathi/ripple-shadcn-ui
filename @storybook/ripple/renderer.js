// renderer.js
export async function renderToCanvas({ storyFn, canvasElement }) {
  console.log("renderer")
  const output = storyFn();

  // Example: simple HTML render
  if (typeof output === 'string') {
    canvasElement.innerHTML = output;
  } else if (output instanceof HTMLElement) {
    canvasElement.innerHTML = '';
    canvasElement.appendChild(output);
  } else {
    canvasElement.innerHTML = `<pre>Unsupported output: ${JSON.stringify(output)}</pre>`;
  }
}
