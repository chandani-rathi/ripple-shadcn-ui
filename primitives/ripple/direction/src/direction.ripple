import { Context } from "ripple"
type Direction = 'ltr' | 'rtl';
const DirectionContext = new Context<Direction | undefined>(undefined);

function useDirection(localDir?: Direction) {
  const globalDir = DirectionContext.get();
  return localDir || globalDir || 'ltr';
}

export {
  useDirection,
  DirectionContext
}