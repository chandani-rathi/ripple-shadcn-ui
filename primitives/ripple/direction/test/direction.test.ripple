import { axe } from 'vitest-axe';
import { useDirection, DirectionContext }  from "../src/direction.ripple";

const TEXT_CHILD = "TEST"

describe("direction", () => {
    beforeEach(() => {
		component RightSideDiv(){
			const direction = useDirection();
			<div dir={direction}>{TEXT_CHILD}</div>
		}
		component Test(){
			DirectionContext.set("rlt")
			<RightSideDiv />
		}
		render(Test);
	});

    it('should have no accessibility violations', async () => {
		expect(await axe(container)).toHaveNoViolations();
	});
    
    it("should work", () => {
         expect(container).toMatchSnapshot()
    })
})