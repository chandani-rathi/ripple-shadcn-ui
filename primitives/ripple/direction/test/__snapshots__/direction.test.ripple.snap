// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`direction > should work 1`] = `
<div>
  <!---->
  <div
    dir="rlt"
  >
    TEST
  </div>
  <!---->
  
</div>
`;
